core/src/commonMain/kotlin/com/ustadmobile/core/controller/ContentEntryEdit2Presenter.kt

```diff
@@ -167,6 +175,21 @@ class ContentEntryEdit2Presenter(context: Any,
 
                         containerUploadManager?.enqueue(uploadJob.cujUid)
                     }
+                }else if (view.selectedUrl != null){
```

Comment:
There is a logical issue here - because there are two separate variables (selectedUrl and selectedFile). The user could select a file, change their mind, and then select a link.

I think it would be better to have a single variable (e.g. selectedUri). Then the logic can check if selectedUri is a file or an http/https link.